apps/studio/src/components/ResourceSelector/ResourceSelector.tsx

```diff
@@ -0,0 +1,141 @@
+import { useState } from "react"
+import {
+  Box,
+  Flex,
+  HStack,
+  Icon,
+  InputGroup,
+  InputLeftElement,
+  Spacer,
+  Text,
+} from "@chakra-ui/react"
+import { Input, Link } from "@opengovsg/design-system-react"
+import fuzzysort from "fuzzysort"
+import { BiHomeAlt, BiLeftArrowAlt, BiSearch } from "react-icons/bi"
+
+import { trpc } from "~/utils/trpc"
+import { ResourceItem } from "./ResourceItem"
+
+interface ResourceSelectorProps {
+  siteId: string
+  selectedResourceId?: string
+  onChange: (resourceId: string) => void
+  isDisabledFn?: (resourceId: string) => boolean
+}
+
+export const ResourceSelector = ({
+  siteId,
+  selectedResourceId,
+  onChange,
+  isDisabledFn,
+}: ResourceSelectorProps) => {
+  const [parentIdStack, setParentIdStack] = useState<string[]>([])
+  const [searchQuery, setSearchQuery] = useState<string>("")
```

Comment:
supernit: don't need type hint
```suggestion
  const [searchQuery, setSearchQuery] = useState("")
```

Reply:
Updated in 7f139851126b5ab4f72d99fbbf9c5588cb2f9adc.